### ISSUE TITLE
Error with large context and AnthropicStream?

### ISSUE THREAD

**Issue by tmyracle** [vercel/ai#383]
I'm doing some testing with feeding larger context into the Anthropic models and I'm running into the following error message:
`error - TypeError: Cannot read properties of undefined (reading 'slice')
    at eval (webpack-internal:///(middleware)/./node_modules/ai/dist/chunk-TWW2ODJW.mjs:17:24)
    at onParse (webpack-internal:///(middleware)/./node_modules/ai/dist/chunk-UJV6VDVU.mjs:28:29)
    at parseEventStreamLine (webpack-internal:///(middleware)/./node_modules/eventsource-parser/dist/index.js:77:9)
    at Object.feed (webpack-internal:///(middleware)/./node_modules/eventsource-parser/dist/index.js:65:7)
    at Object.transform (webpack-internal:///(middleware)/./node_modules/ai/dist/chunk-UJV6VDVU.mjs:37:14)
    at g (evalmachine.<anonymous>:102:35)
    at w (evalmachine.<anonymous>:107:14)
    at Object.transform (evalmachine.<anonymous>:1953:33)
    at TransformStreamDefaultController.<anonymous> (evalmachine.<anonymous>:2063:49)
    at pr (evalmachine.<anonymous>:2181:15)`

My edge handler is pretty basic, so I don't think its anything weird there plus it works fine with smaller requests (abbreviated code path):
```
      const altPrompt = `
        ${userPrompt}

        Documents to reference: \n${combinedText}
      `;

      const anthropicResponse = await fetch(
        "https://api.anthropic.com/v1/complete",
        {
          method: "POST",
          headers: {
            "Content-Type": "application/json",
            "x-api-key": process.env.ANTHROPIC_API_KEY || "",
          },
          body: JSON.stringify({
            prompt: `\n\nHuman: ${altPrompt}\n\nAssistant:`,
            model: "claude-2",
            max_tokens_to_sample: 10000,
            temperature: 0.0,
            stream: true,
          }),
        }
      );

      stream = AnthropicStream(anthropicResponse);

     return new StreamingTextResponse(stream);
```

There doesn't seem to be an issue with smaller context requests to the endpoint so not sure if something is timing out or what is happening. I've tested the requests in a jupyter notebook hitting the anthropic API directly and I get valid responses so doesn't look like its a server error from Anthropic's side and I'm still below the context window limit. Any tips or help would be greatly appreciated and thanks to everyone who works on this repo!

**Comment by youngbloodcyb** [comment]
I just came on to post this issue as well. I'm getting the exact same error and experiencing the same thing. It seems to break with larger context.

**Comment by MaxLeiter** [comment]
Thanks for the report. I think we need to handle `ping` and `error` events in our AnthropicStream. 

[Anthropic docs](https://docs.anthropic.com/claude/reference/streaming#events)
[Our code](https://github.com/vercel-labs/ai/blob/main/packages/core/streams/ai-stream.ts#L44-L61)

A PR would be much appreciated, otherwise I will get to it when I can.

**Comment by MaxLeiter** [comment]
hey @tmyracle, I've been unable to reproduce the original error on my end. Can you provide a more complete example (with a prompt that fails) or give #388 a try?

**Comment by NiluK** [comment]
@MaxLeiter, this is a pretty easy fairly constant replication when you ask claude to output a large chunk of text. here is a sample prompt:

``${HUMAN_PROMPT} 
  
1. You are an expert in Twine, a tool for creating interactive stories. You are using a user provided storyline, indicated with <storyline> and the user provided lore indicated with <lore>. You must use the lore to create the story.
2. Each Twee passage starts with :: Passage Name [Passage Number] (X,Y)
3. You are using a user provided storyline, indicated with <storyline> and the user provided lore indicated with <lore>. You must use the lore to create the story.
4. When creating this story, you must explicitly reference the provided lore by mentioning specific technologies, services, companies, models, principles, or other concepts from the lore. These references should be woven throughout the story seamlessly and help build the world. Do not just mention the lore in passing - use it to enrich the narrative and demonstrate knowledge of the details
5. The story must have at least ten passages and multiple branching choices.
6. Links between passages should be created using the [[choice->name]] syntax. 
7. Ensure at least once link is always created to the next passage if it's not a branching choice (e.g. [[continue->next passage]]).

This is the provided lore: 
<lore>

<lore>
1. The Black City No traveler to the Fade can fail to spot the Black City. It is one of the few constants of that everchanging place. No matter where one might be, the city is visible. (Always far off, for it seems that the only rule of geography in the Fade is that all points are equidistant from the Black City.) The Chant teaches that the Black City was once the seat of the Maker, from whence He ruled the Fade, left empty when men turned away from Him. Dreamers do not go there, nor do spirits. Even the most powerful demons seem to avoid the place. It was golden and beautiful once, so the story goes, until a group of powerful magister-lords from the Tevinter Imperium devised a means of breaking in. When they did so, their presence defiled the city, turning it black. (Which was, perhaps, the least of their worries.) —From Beyond the Veil: Spirits and Demons by Enchanter Mirdromel Blood Magic: The Forbidden School 
2. Foul and corrupt are you Who have taken My gift And turned it against My children. —Transfigurations 18:10. The ancient Tevinters did not originally consider blood magic a school of its own. Rather, they saw it as a means to achieve greater power in any school of magic. The name, of course, refers to the fact that magic of this type uses life, specifically in the form of blood, instead of mana. It was common practice, at one time, for a magister to keep a number of slaves on hand so that, should he undertake the working of a spell that was physically beyond his abilities, he could use the blood of his slaves to bolster the casting. Over time, however, the Imperium discovered types of spells that could only be worked by blood. Although lyrium will allow a mage to send his conscious mind into the Fade, blood would allow him to find the sleeping minds of others, view their dreams, and even influence or dominate their thoughts. Just as treacherous, blood magic allows the Veil to be opened completely so that demons may physically pass through it into our world. The rise of the Chant of Light and the subsequent fall of the old Imperium has led to blood magic being all but stamped out—as it should be, for it poses nearly as great a danger to those who would practice it as to the world at large. —From The Four Schools: A Treatise, by First Enchanter Josephus. 
3. Maleficarum It has been asked, "What are maleficarum? How shall we know them?" I have been as troubled by this question as you. You have come to me for the wisdom of the Maker, but none have seen the Maker's heart save Beloved Andraste. And so I have done as all mortals must, and looked to the words of His prophet for answers. And there, I found respite from a troubled mind. For she has said to us, "Magic exists to serve man, and never to rule over him." Therefore, I say to you, they who work magic which dominates the minds and hearts of others, they have transgressed the Makers law. Also, Our Lady said to us, "Those who bring harm without provocation to the least of His children are hated and accursed by the Maker." And so it is made clear to me, as it should be to us all: That magic which fuels itself by harming others, by the letting of blood, is hated by the Maker. Those mages who honor the Maker and keep His laws we welcome as our brothers and sisters. Those who reject the laws of the Maker and the words of His prophet are apostate. They shall be cast out, and given no place among us. —From The Sermons of Justinia I. </lore>

</lore>

This is the provided storyline 

<storyline>
The aging patriarch of an organized crime dynasty transfers control of his clandestine empire to his reluctant son.
</storyline>

Review the passages provided. Count the number of passages and ensure it is at least 10. Edit the story to seamlessly integrate those details;
Review the links between passages. Ensure there is at least one link between passages. If there are no links, add at least one link between passages;
Review the lore provided. List any details you did not properly incorporate or reference. Edit the story to seamlessly integrate those details
Once you have completed the at least 10 passage story integrating the storyline and lore, end the story without any commentary or review summary

${AI_PROMPT} Here is a story that is at least 10 passages and multiple branching choices using the provided storyline and lore:

:: Start (0,0)
`;

This is I believe due to. a ping event, which has no data, a simple guard should fix this. I will add a PR

**Comment by MaxLeiter** [comment]
Fixed by #448

**Comment by lgrammel** [comment]
## Identify issue type

Identified as **Bug**  
Confidence: **High**

### Rationale
This is a concrete runtime failure in existing AnthropicStream behavior caused by not handling valid Anthropic stream events.

### Evidence
- AnthropicStream throws `TypeError: Cannot read properties of undefined (reading 'slice')` during larger streamed requests.
- Another user reproduced the same failure, and a maintainer identified unhandled Anthropic `ping` and `error` stream events as the likely cause.
- A reproducible prompt was provided, with the failure attributed to ping events lacking data.
- The maintainer confirmed the issue was fixed by PR #448.
